inc/js/expiring-posts.js

```diff
@@ -7,12 +7,13 @@ jQuery(document).ready( function($) {
 		return;
 
 	var $postStatus = $( document.getElementById( 'post_status' )),
-			saveButton = document.getElementById('save-post');
+		saveButton = document.getElementById('save-post'),
+		originalStatus = $postStatus.val();
 
 	// JS hack to append Expired to the post status editor
 	$postStatus.append( $( document.getElementById( 'expired-status' ) ) );
 	// reset post_status value after appending expired option
-	$postStatus.val( AdminExpiringPosts.post_status );
+	$postStatus.val( originalStatus );
```

Comment:
If you edited a post that was expired already `originalStatus` would be `pending`. The `#expired-status` `<option>` that is being appended on is already `selected=selected`. This line 16 then attempts to set the select the value `pending`, which selects Pending. It should be setting to `expired`. A conditional should be added to see if `originalStatus` should be changed to `expire` before setting the value.